cmd/tasks_run.go

```diff
@@ -42,7 +43,29 @@ var tasksPreRun = &cobra.Command{
 			return err
 		}
 		fmt.Println("Executing Pre-rollout Tasks")
-		err = runTasks(iterateTaskGenerator(true, runCleanTaskInEnvironment, buildValues, true), lYAML.Tasks.Prerollout, lagoonConditionalEvaluationEnvironment)
+
+		// We actually want to unidle the namespace before running pre-rollout tasks,
+		// so we wrap the usual task runner before calling it
+		unidleThenRun := func(namespace string, incoming lagoon.Task) error {
+			err := lagoon.UnidleNamespace(context.TODO(), namespace, namespaceUnidleWaitInSeconds, namespaceUnidleWaitInSeconds)
+			if err != nil {
+				switch {
+				case errors.Is(err, lagoon.NamespaceUnidlingTimeoutError):
+					fmt.Println("Namespace unidling is taking longer than expected - this might affect pre-rollout tasks that rely on multiple services")
+				default:
+					return err
```

Comment:
I'm a bit nervous about this one, basically if there was an error talking to kubernetes here, that will fail the pre-rollout tasks and return whatever error is presented to the user?

If that is the case, could we at least wrap the error in something that makes it clear that there was something unrelated to their codebase that caused the unidling to fail?

Reply:
This would also block the task from running?